bids/layout/index.py

```diff
@@ -164,10 +165,37 @@ def index_files(self):
         """Index all files in the BIDS dataset. """
         self._index_dir(self.root, self.config)
 
-    def index_metadata(self):
+    def index_metadata(self, **filters):
         """Index metadata for all files in the BIDS dataset. """
+        if filters:
+            default_ext = ['[.]+']
+            # ensure we are returning objects
+            filters['return_type'] = 'object'
+            # until 0.11.0, user can specify extension or extensions
+            if filters.get('extension'):
```

Comment:
Simplify this a bit. I was going to suggest just renaming `extensions` to `extension` if you found it to simplify it further, but then you wouldn't get the `DeprecationWarning` that `.get` issues.

```suggestion
            ext_key = 'extensions' if 'extensions' in filters else 'extension'

            if not filters.get(ext_key):
                default_ext = ['[.]+']
```